bouncerscript/task.py

```diff
@@ -140,21 +146,43 @@ def check_location_path_matches_destination(product_name, path):
         raise ScriptWorkerTaskException(err_msg)
 
 
-def check_versions_are_successive(current_version, payload_version):
-    current_bouncer_version = GeckoVersion.parse(current_version)
-    candidate_version = GeckoVersion.parse(payload_version)
-
-    if current_bouncer_version.major_number == candidate_version.major_number:
-        err_msg = ("At this point, in-tree version can't be equal to bouncer "
-                   "counterpart".format(payload_version, current_version))
-        raise ScriptWorkerTaskException(err_msg)
-    elif current_bouncer_version.major_number > candidate_version.major_number:
-        err_msg = ("In-tree version {} can't be less than current bouncer "
-                   "counterpart".format(payload_version, current_version))
-        raise ScriptWorkerTaskException(err_msg)
-    elif (candidate_version.major_number - current_bouncer_version.major_number) > 1:
-        err_msg = ("In-tree version {} can't be greater than current bouncer "
-                   "by more than 1 digit".format(payload_version, current_version))
+def check_versions_are_successive(current_version, payload_version, product):
+    """Function to check if the provided version in the payload and the existing
+    one in bouncer are successive as valid versions."""
+    def _successive_sanity(current_identifier, candidate_identifier):
```

Comment:
@JohanLorenzo As a follow-up in the future, I'd like to move operations such as this one directly within `mozilla-version` repo so that we centralize and reuse code. What do you think?

Reply:
Yeah, we could have something like:

```python
next_version = current_version.bump_major()
if candidate != next_version:
     raise ScriptWorkerTaskException()
```

in bouncerscript. That would be simpler!

Reply:
Agreed, this would be awesome! 👍 I filed https://github.com/mozilla-releng/mozilla-version/issues/62 to track this  👍